### PR TITLE
add `order_by` and `order_how` query params

### PR DESCRIPTION
**Pull request by beav**

### CODE REVIEW

File: system_baseline/views/v0.py [hunk 1]
```diff
@@ -131,9 +157,27 @@ def delete_baselines_by_ids(baseline_ids):
...
+    """
+    helper method to set ordering on query
+    """
+    if order_by == "display_name":
+        if order_how == "ASC":
+            query = query.order_by(SystemBaseline.display_name.asc())
+        else:
+            query = query.order_by(SystemBaseline.display_name.desc())
+    elif order_by == "created_on":
+        if order_how == "DESC":
```
Comment (ehelms) [inline system_baseline/views/v0.py]: Minor, but would be nice if this matched the previous if-block by starting with ASC

File: system_baseline/views/v0.py [hunk 2]
```diff
@@ -131,9 +157,27 @@ def delete_baselines_by_ids(baseline_ids):
...
 
 
+def _create_ordering(order_by, order_how, query):
+    """
+    helper method to set ordering on query
+    """
+    if order_by == "display_name":
+        if order_how == "ASC":
+            query = query.order_by(SystemBaseline.display_name.asc())
+        else:
```
Comment (ehelms) [inline system_baseline/views/v0.py]: Does something gate the value of `order_how` ? The naked else implies no matter what I pass for `order_how` I will get a descending sort.

File: system_baseline/views/v0.py (hunk 1 above)
Comment (beav) [reply system_baseline/views/v0.py]: missed this before, thx

File: system_baseline/views/v0.py (hunk 2 above)
Comment (beav) [reply system_baseline/views/v0.py]: its gated to either DESC or ASC in the openapi yaml. However I rewrote it a bit to make it clearer elsewhere, will update here.

**Comment by beav** [comment]
I added a second commit for search by display_name. if I created both PRs off master, there would have been a lot of conflicts so it was easier to just add to here.

File: system_baseline/openapi/api.spec.yaml [hunk 3]
```diff
@@ -297,6 +302,35 @@ components:
...
+          - ASC
+          - DESC
+      description: Direction of the ordering, defaults to ASC
+    displayNameParam:
+      name: display_name
+      in: query
+      required: false
+      schema:
+        $ref: "#/components/schemas/DisplayName"
+      description: Direction of the ordering, defaults to ASC
```
Comment (ehelms) [inline system_baseline/openapi/api.spec.yaml]: Needs an update

File: system_baseline/openapi/api.spec.yaml [hunk 4]
```diff
@@ -297,6 +302,35 @@ components:
...
+      required: false
+      schema:
+        default: "ASC"
+        type: string
+        enum:
+          - ASC
+          - DESC
+      description: Direction of the ordering, defaults to ASC
+    displayNameParam:
+      name: display_name
```
Comment (ehelms) [inline system_baseline/openapi/api.spec.yaml]: Is the idea to start simple with specific search fields vs. having a generic search field that can accept different search types? e.g. `search='display_name = foo'` `'search='display_name ~ foo'`

Does the API specification for platform talk about search?

File: system_baseline/views/v0.py [hunk 5]
```diff
@@ -131,23 +157,52 @@ def delete_baselines_by_ids(baseline_ids):
...
-def get_baselines(limit, offset):
+def get_baselines(limit, offset, order_by, order_how, display_name=None):
     """
     return a list of baselines given their ID
     """
     account_number = view_helpers.get_account_number(request)
     query = SystemBaseline.query.filter(SystemBaseline.account == account_number)
 
+    if display_name:
+        query = query.filter(SystemBaseline.display_name == display_name)
```
Comment (ehelms) [inline system_baseline/views/v0.py]: Is this an exact match search? How would I find all baselines with X in the name?

File: system_baseline/openapi/api.spec.yaml (hunk 4 above)
Comment (beav) [reply system_baseline/openapi/api.spec.yaml]: its currently exact only, but I'm guessing a library exists already to do more complex searches. I'll take a look tomorrow

File: system_baseline/views/v0.py (hunk 5 above)
Comment (beav) [reply system_baseline/views/v0.py]: I altered the search to be "contains" instead of exact match

**Comment by beav** [comment]
@ehelms I took a look at a couple of libraries that allow complex search queries. However they are pretty heavyweight and would require a service rewrite.

For now I altered the search so if a substring is in a display_name, it is returned. This matches what inventory service does.

**Pull request merged**